packages/expo-dev-launcher/bundle/native-modules/DevLauncherInternal.ts

```diff
@@ -24,6 +24,30 @@ export async function loadUpdate(updateUrl: string, projectUrl: string) {
   return await DevLauncher.loadUpdate(updateUrl, projectUrl);
 }
 
+export async function getNavigationStateAsync() {
+  return await DevLauncher.getNavigationState();
+}
+
+export async function consumeNavigationStateAsync() {
+  const serializedNavigationState = await DevLauncher.getNavigationState();
+  let navigationState;
+
+  try {
+    navigationState = JSON.parse(serializedNavigationState);
+  } catch (error) {}
+
+  clearNavigationState();
+  return navigationState;
+}
+
+export async function saveNavigationState(navigationState: string) {
+  return await DevLauncher.saveNavigationState(navigationState);
+}
+
+export async function clearNavigationState() {
```

Comment:
naming of the exported functions in this file is a little inconsistent -- some have `Async` suffix but some don't. not a huge deal but wouldn't hurt to standardize